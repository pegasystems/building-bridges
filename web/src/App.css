* {
    margin: 0;
    padding: 0;
}

a:link {
    color: #386786
}

a:visited {
    color: #5a598c
}

a.title {
    color: #FFF;
}

section,
header,
aside {
    display: block;
}

.App {
  font: 15px/1.3 'Open Sans', Arial, Helvetica, sans-serif;
  background-color: #E7E7E7;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.title {
    text-decoration: none;
    font-weight: 300;
    color: #FFF;
    font-family: 'Open Sans Condensed', Arial, Helvetica, sans-serif;
}

.title-div {
    text-align: center;
}

.App-link {
  color: #61dafb;
}

.creation-flex {
    display: flex;
    justify-content: space-between;
    width: auto;
    padding: 6px 0;
}

.creation-flex .survey-name {
    width:100%
}

.creation-flex * {
    margin: 0 6px;
    /*flex-grow: 1;*/
}

.options{
    padding: 10px;
}


/* The container */
.container-checkbox {
    display: block;
    position: relative;
    padding-left: 35px;
    margin-bottom: 12px;
    cursor: pointer;
    font-size: 17px;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
}

/* Hide the browser's default checkbox */
.container-checkbox input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
}

/* Create a custom checkbox */
.container-checkbox .checkmark {
    position: absolute;
    top: 0;
    left: 0;
    height: 25px;
    width: 25px;
    background-color: #eee;
}

/* On mouse-over, add a grey background color */
.container-checkbox:hover input ~ .checkmark {
    background-color: #ccc;
}

/* When the checkbox is checked, add a blue background */
.container-checkbox input:checked ~ .checkmark {
    background-color: #2196F3;
}

/* Create the checkmark/indicator (hidden when not checked) */
.container-checkbox .checkmark:after {
    content: "";
    position: absolute;
    display: none;
}

/* Show the checkmark when checked */
.container-checkbox input:checked ~ .checkmark:after {
    display: block;
}

/* Style the checkmark/indicator */
.container-checkbox .checkmark:after {
    left: 9px;
    top: 5px;
    width: 5px;
    height: 10px;
    border: solid white;
    border-width: 0 3px 3px 0;
    -webkit-transform: rotate(45deg);
    -ms-transform: rotate(45deg);
    transform: rotate(45deg);
}



/************************

 TO CLEAN

*************************/


body {
    font: 15px/1.3 'Open Sans', Arial, Helvetica, sans-serif;
    background-color: #E7E7E7;
}

a:link {
    color: #386786
}

a:visited {
    color: #5a598c
}

a.title {
    color: #FFF;
}

section,
header,
aside {
    display: block;
}


/*-------------------------
	Header & Titles
--------------------------*/

header {
    color: #666;
    background-color: #151515;
    font-size: 1.3em;
    line-height: 1.5;
    padding: 20px 0;
    text-align: center;
    position: relative;
}

.title {
    text-decoration: none;
    font-weight: 300;
    color: #FFF;
    font-family: 'Open Sans Condensed', Arial, Helvetica, sans-serif;
}

.title-div {
    text-align: center;
}

.survey-title {
    text-decoration: none;
    margin: 1% 0 0.5% 0;
    font-weight: 800;
    color: rgb(0, 0, 0);
    font-family: 'Open Sans Condensed', Arial, Helvetica, sans-serif;
}

.survey-list td:first-of-type {
    width: 100%;
}

.fixed {
    min-width: 62px;
    max-width: 25%;
}

.aligned-center {
    text-align: center;
}

.results-info {
    background-color: #d7d7d7;
    border: 1px solid #e2e2e2;
    border-radius: 6px 6px 6px 6px;
    padding: 16px;
    margin: 0 auto;
    width: 63%;
    text-align: left;
}

.results-stats-table {
    border: 1px solid black;
    margin: auto;
    border-collapse: collapse;
    padding: 6px;
}

.survey-description {
    text-align: center;
    text-decoration: none;
    margin-top: 0.5%;
    font-weight: 400;
    color: rgb(0, 0, 0);
    font-family: 'Open Sans Condensed', Arial, Helvetica, sans-serif;
}

h1 .beta {
    font-size: 0.3em;
    text-transform: uppercase;
    border-radius: 3px;
    background-color: #a72b0b;
    padding: 2px;
    color: #e1e1e1;
}

footer {
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 1em 0;
}

footer div {
    position: relative;
    display:inline-block;
    background-color: #656860;
    margin: auto;
    max-width: 40%;
    border-radius: 25px;
    color: #e1e1e1;
    text-align: left;
    padding: 1.5em;
}

footer a  {
    color: #2FA3D1;
}


/*-------------------------
	Question List
--------------------------*/

ul, table {
    width: 50%;
    margin: 30px auto;
    counter-reset: ranks;
}

ul li {
    padding: 20px;
    background-color: #f9f9f9;
    border: 1px solid #eeeeee;
    border-radius: 6px;
    list-style: none outside none;
    margin: 6px 0 0 0;
    position: relative;
    text-shadow: 1px 1px 0 white;
    min-height: 38px;
}

table {
    border-radius: 6px;
}

table thead,tbody {
    max-width: 40%;
    padding: 20px;
    background-color: #f9f9f9;
    border-radius: 6px 6px 6px 6px;
    list-style: none outside none;
    margin: 6px 0 0 30px;
    position: relative;
    min-height: 38px;
}

#share_container{
    min-height: 20px;
}

.survey-list {
    width: 100%;
    margin: 30px auto;
    max-width: 700px;
    text-align: left;
}

.survey-list td,th {
    padding: 12px;
}

table .new {
    animation: new;
    animation-duration: 2s;
    animation-timing-function: ease;
}

@keyframes new {
    from {background-color: darkkhaki}
    to {background-color: #f9f9f9}
}

.copy {
    text-decoration: underline;
    cursor: copy;
}

.question-box {
    background-color: #f9f9f9;
    border: 1px solid #eeeeee;
    border-radius: 6px 6px 6px 6px;
    padding: 12px;
    margin: 0 auto;
    width: 65%;
    text-align: left;
}

.question-box form {
    padding: 6px 0;
}

.admin-box {
    text-align: center;
    width: 100%;
    padding: 12px;
}

.question-box textarea {
    width: 100%;
    font-family: inherit;
    height: 100px;
    font-size: 17px;
    box-sizing: border-box;
    padding: 7px;
}

.creation-flex {
    display: flex;
    justify-content: space-between;
    width: auto;
    padding: 6px 0;
}

.creation-flex .survey-name {
    width:100%
}

.creation-flex * {
    margin: 0 6px;
    /*flex-grow: 1;*/
}

.submit-up,
/*.submit-down {
    background: url('../img/sprite.svg') no-repeat;
}*/

.submit-up {
    color: #90d184;
    background-position: 5px -300px;
}

.up {
    border-color: rgba(144, 209, 132, 1);
}

.down .submit-up {
    opacity: .3;
}

.submit-down {
    color: #e67a85;
    top: 82px;
    background-position: 5px -192px;
}

.up .submit-down {
    opacity: .3;
}

.down {
    border-color: rgba(230, 122, 133, 1);
}

.permalink {
    background: #2FA3D1;
    position: absolute;
    left: -28px;
    top: 0.7em;
    width: 28px;
    height: 28px;
    padding: 3px 0 0;
    text-align: center;
    color: #fff;
    font-size: 20px;
    text-decoration: none;
    text-shadow: none;
    display: none;
}

input.remove[type=submit] {
    background:#d13240;
    border:0 none;
    cursor:pointer;
    border-radius: 5px;
    position: absolute;
    right: -28px;
    width: 28px;
    height: 28px;
    text-align: center;
    color: #fff;
    font-size: 20px;
    text-decoration: none;
    text-shadow: none;
    display: none;
}

.hide-eye {
    border-radius: 5px;
    right: -256px;
    font-size: 20px;
    position: relative;
    top: -0.5em;
}

li:hover input.remove[type=submit] {
    display: block;
}

li:hover .hide-eye {
    display: block;
}


li:hover .permalink {
    display: block;
}

header:before {
    content: "";
    position: absolute;
    z-index: 1;
    top: 0;
    right: 83%;
    bottom: 0;
    left: 0;
    border-top: 98px solid #2FA3D1;
    border-right: 27px solid transparent;
}

.question-box form input {
    margin-top: 5px;
}

h4 {
    font-family: 'Open Sans Condensed', Arial, Helvetica, sans-serif;
    font-size: 22px;
    font-weight: normal;
}

ul {
    width: 65%;
    margin: 30px auto;
    counter-reset: ranks;
}

ul li.down:before {
    color: rgba(230, 122, 133, 1);
}

ul li.up:before {
    color: rgba(144, 209, 132, 1);
}

@media (max-width: 600px) {
    ul {
        width: 95%;
    }
    ul li {
        min-height: 75px;
    }
}

@media only screen and (max-width: 1080px) {
    ul {
        max-width: 80%;
    }
}

@media only screen and (max-width: 990px) {
    ul {
        max-width: 90%;
    }
}

li:target {
    -webkit-animation: target 2s;
    /* Chrome, Safari, Opera */
    -moz-animation: target 2s;
    animation: target 2s;
}


/* Chrome, Safari, Opera */

@-webkit-keyframes target {
    from {
        background: #FFD268;
    }
    to {
        background: #F9F9F9;
    }
}


/* Standard syntax */

@keyframes target {
    from {
        background: #FFD268;
    }
    to {
        background: #F9F9F9;
    }
}


/*---------------
	styles for what is this page -- I should really make this ONLY load when you are on /what
------------------*/

.what-container {
    font: large/1.556 "Libertine", Palatino, "Palatino Linotype", "Book Antiqua", Georgia, "Times New Roman", serif;
    word-spacing: 0.1em;
    max-width: 960px;
    margin: 0 auto;
    padding: 2em;
    background-color: #FFF;
}

.what-container>.info {
    padding: .25em .25em .25em 0;
}

.what-container>p {
    padding-left: .4em;
    margin: 1em 0;
}
