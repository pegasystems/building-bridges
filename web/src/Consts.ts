export const SURVEYS_API = '/api/surveys/';
export const QUESTIONS_ENDPOINT = '/questions';