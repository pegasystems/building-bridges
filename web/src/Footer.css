.App-footer {
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 1em 0;
}

.App-footer div {
    position: relative;
    display:inline-block;
    background-color: #656860;
    margin: auto;
    max-width: 40%;
    border-radius: 25px;
    color: #e1e1e1;
    text-align: left;
    padding: 1.5em;
}

.App-footer a  {
    color: #2FA3D1;
}