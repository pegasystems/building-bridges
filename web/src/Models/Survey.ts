export default class Survey {
    key!: string;
    title!: string;
    results_secret!: string;
    admin_secret!: string;
    viewsNumber: number;
    votersNumber: number;
    questionersNumber: number;
    date!: string;


    constructor(key: string, title: string, results_secret: string, admin_secret: string, viewsNumber: number, votersNumber: number, commentatorsNumber: number) {
        this.key = key;
        this.title = title;
        this.results_secret = results_secret;
        this.admin_secret = admin_secret;
        this.viewsNumber = viewsNumber;
        this.votersNumber = votersNumber;
        this.questionersNumber = commentatorsNumber;
        this.date = new Date().toString();
    }

    getUrl() {
        return `surveys/${this.key}`;
    }
}