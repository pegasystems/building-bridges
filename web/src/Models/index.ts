export {default as Survey} from "./Survey"
export {default as Question} from "./Question"
export {UserVote} from "./UserVote"