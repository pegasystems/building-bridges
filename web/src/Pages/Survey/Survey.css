.qa-box {
	display: flex;
	flex-direction: column;
}
.center {
    flex: 1;
    order: 0;
}

.footnote {
	font-style: italic;
	font-size: 13px;
	color: #949494;
	margin-top: 2%;
}

.right {
    order: 1;
    margin-left: 15px;
    flex-direction: column;
	justify-content: right;
	flex-wrap: wrap;
}

.thumbs {
	display: flex;
	flex-wrap: nowrap;
	justify-content: space-between;
}

.thumbs > div[class=item] {
	flex-basis: 30%;
}

.vote {
	display: flex;
}

#voted .right{
	flex-basis: 190px;
}

.vote form {
	float: right;
}

.vote > .right form input {
	line-height: 40px;
	font-weight: bold;
	font-size: 24px;
	padding-left: 40px;
	border: medium none;
	border-spacing: 0;
}

ul li {
	padding: 15px 15px 15px 10px;
}

ul li:before {
    float: left;
    font-size: 26px;
    color: #aaa;
	width: 50px;
}

ul.enumerate li:before {
    counter-increment: ranks;
    content: counter(ranks);
}

h2 {
    width: 65%;
}

svg {
	display: table;
	margin: 0 auto;
}

.voted-unvoted-header{

    width: 66%;
    margin: 18px auto;

}
.hide-votes > .responses {
	display:table;
	width: 100%;
}

.hidden {
	opacity: 0.2;
}

.hide-votes > .responses > .questions {
	display: table-cell;
	width: 50%;
}

.hide-votes > .responses > .questions > ul {
	width: 90%;
}

.hide-votes > .responses > .questions > .voted-unvoted-header{
	width: 90%;
}

@media only screen and (max-width: 800px) {
	.hide-votes > .responses {
		display:block;
		width: inherit;
	}

	.hide-votes > .responses > .questions {
		display: block;
		width: inherit;
	}

	.hide-votes > .responses > .questions > ul {
		width: 65%;
	}

	.hide-votes > .responses > .questions > .voted-unvoted-header{
		width: 65%;
	}
}

@media only screen and (max-width: 600px) {

	.hide-votes > .responses > .questions > ul {
		width: 90%;
	}

	.hide-votes > .responses > .questions > .voted-unvoted-header{
		width: 90%;
	}
}