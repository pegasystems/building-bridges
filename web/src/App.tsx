import React from 'react';
import './App.css';

import Header from './Header';
import Footer from './Footer';
import Home from './Pages/Home/Home';
import Survey from './Pages/Survey/Survey';
import Page404 from './Pages/Page404';

import {
  BrowserRouter as Router,
  Switch,
  Route
} from 'react-router-dom';

export default function App() {
  return (
    <div className="App">
      <Header/>
      <Router>
        <div>
          <Switch>
            <Route path="/surveys/:surveyKey" component={Survey}/>
            <Route exact path="/" component={Home}/>
            <Route path="page404" component={Page404}/>
            <Route component={Page404}/>
          </Switch>
        </div>
      </Router>
      <Footer/>
    </div>
  );
}
