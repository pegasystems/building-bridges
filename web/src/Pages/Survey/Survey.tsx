import React from 'react';
import SurveyLink from '../../Components/SurveyLink';
import queryString from 'query-string';
import update from 'immutability-helper';
import Loader from 'react-loader-spinner';
import QuestionList from '../../Components/QuestionList';
import NewQuestionBox from '../../Components/NewQuestionBox';
import SurveyResultsInfo from '../../Components/SurveyResultsInfo';
import {SURVEYS_API, QUESTIONS_ENDPOINT} from '../../Consts'
import {shuffle} from '../../utils'
import * as Models from '../../Models';

import './Survey.css';

interface SurveyState {
    questions: Models.Question[];
    title: string;
    description: string;
    fetchInProgress: boolean;
    hideVotes: boolean;
    viewsNumber: number;
    questionersNumber: number;
    votersNumber: number;
    date: string;
    open: boolean;
}

interface SurveyProps {
    match: any;
    location: any;
}

export default class Survey extends React.Component<SurveyProps, SurveyState> {
    surveyKey: string;
    results_secret?: string | string[] | null;
    admin_secret?: string;

    constructor(props: SurveyProps) {
        super(props);
        this.state = {
            questions: [] as Models.Question[],
            fetchInProgress: true,
            title: '',
            description: '',
            hideVotes: false,
            viewsNumber: 0,
            questionersNumber: 0,
            votersNumber: 0,
            date: '',
            open: true
        };
        this.surveyKey = props?.match?.params?.surveyKey;
        this.results_secret = queryString.parse(props?.location?.search).results_secret?.toString();
        this.admin_secret = queryString.parse(props?.location?.search).admin_secret?.toString()
    }

    componentDidMount() {
        this.getQuestions();
    }

    getQuestions() {
        var fetchUrl = `${SURVEYS_API}${this.surveyKey}`

        if(this.results_secret) {
            fetchUrl += `?results_secret=${this.results_secret}`
        }
        if(this.admin_secret) {
            fetchUrl+=`${this.results_secret? '&' : '?'}admin_secret=${this.admin_secret}`
        }

        fetch(fetchUrl)
            .then(response => {
                if (response.status === 404 || response.status === 401) {
                    window.location.href = '/page404';
                }
                return response.json();
            }).then(data => {
            data.questions = shuffle(data.questions);
            data.questions = data.questions.map((questionDict: any) => {
                return Models.Question.createQuestionFromApiResult(questionDict);
            })
            this.setState({...data, ...{fetchInProgress: false}});
            if (window.location.hash) {
                window.location.href = window.location.hash;
            }
        });
    }

    deleteQuestionCallback = (_: string) => {
        this.getQuestions();
    }

    addQuestionToState = (question: Models.Question) => {
        if (this.state.hideVotes) {
            question.hideVotes = true;
        }
        this.setState(state => {
            return {questions: [question, ...state.questions]};
        });
        window.location.href = `#${question._id}`;
    }

    addVoteCallback = (question: Models.Question, voteType: Models.UserVote) => {
        if (this.state.hideVotes) {
            question.hideVotes = false;
            fetch(`${SURVEYS_API}${this.surveyKey}${QUESTIONS_ENDPOINT}/${question._id}`, {
                method: 'GET',
                headers: {
                    'Content-Type': 'application/json'
                }
            })
                .then(response => response.json())
                .then(data => {
                    this.updateQuestionInState(data);
                });
        } else {
            question.addUserVote(voteType);
            this.updateQuestionInState(question);
        }
    }

    updateQuestionInState = (question: Models.Question) => {
        const questionIndex = this.state.questions.findIndex(q => q._id === question._id);
        this.setState({
            questions: update(this.state.questions, {
                [questionIndex]:
                    {$set: question}
            })
        });
    }

    updateSurveyStateCallback = (newState: boolean) => {
        this.setState({open: newState})
    }

    deleteVoteCallback = (question: Models.Question) => {
        let questionModel = Models.Question.createQuestionFromApiResult(question);
        if (this.state.hideVotes && questionModel.read === "false") {
            questionModel.hideVotes = true;
        } else {
            questionModel.removeUserVote();
        }
        questionModel.voted = Models.UserVote.None;
        this.updateQuestionInState(questionModel);
    }

    markAsReadCallback = (question: Models.Question) => {
        if (this.state.hideVotes) {
            question.hideVotes = false;
            fetch(`${SURVEYS_API}${this.surveyKey}${QUESTIONS_ENDPOINT}/${question._id}`, {
                method: 'GET',
                headers: {
                    'Content-Type': 'application/json'
                }
            })
                .then(response => response.json())
                .then(data => {
                    question.toggleRead();
                    if (question.read === "false" && question.voted === Models.UserVote.None) {
                        data.upvotes = null
                    }
                    let newQuestionState = Models.Question.createQuestionFromApiResult(data);
                    this.updateQuestionInState(newQuestionState);
                });
        } else {
            question.toggleRead();
            this.updateQuestionInState(question);
        }
    }

    getQuestionsLists() {
        const questionFunctions = {
            deleteQuestion: this.deleteQuestionCallback,
            addVoteCallback: this.addVoteCallback,
            deleteVoteCallback: this.deleteVoteCallback,
            markAsReadCallback: this.markAsReadCallback,
            updateQuestionInState: this.updateQuestionInState
        };

        let questionsLists;
        if (this.showSingleColumn()) {
            const header = !this.showNewQuestionBox() ? 'Results' : '';
            questionsLists =
                <QuestionList questions={this.state.questions}
                              functions={questionFunctions}
                              header={header}
                              enumerate={true}
                              sort={true}
                              voted={true}
                              surveyKey={this.surveyKey}
                              isResultPage={this.showResultsSummary()}
                              adminSecret={this.admin_secret}
                              open={this.state.open}
                />;
        } else {
            questionsLists =
                <>
                    <QuestionList
                        questions={this.state.questions.filter(q => !q.hideVotes)}//q => q.voted !== Models.UserVote.None)}
                        functions={questionFunctions}
                        sort={true}
                        header="Voted questions"
                        voted={true}
                        enumerate={true}
                        surveyKey={this.surveyKey}
                        isResultPage={false}
                        adminSecret=''
                        open={this.state.open}
                    />

                    <QuestionList
                        questions={this.state.questions.filter(q => q.hideVotes)}//q => q.voted === Models.UserVote.None)}
                        functions={questionFunctions}
                        sort={false}
                        header="Unvoted questions"
                        voted={false}
                        enumerate={false}
                        surveyKey={this.surveyKey}
                        isResultPage={false}
                        adminSecret=''
                        open={this.state.open}
                    />
                </>;
        }
        return questionsLists;
    }

    private showSingleColumn() {
        return this.admin_secret || this.results_secret || !this.state.open || !this.state.hideVotes;
    }

    render(): JSX.Element {
        return (
            <div className={(this.state.hideVotes && this.state.open) ? 'hide-votes' : 'not-hide'}>
                <div className="title-div">
                    <h1 className="survey-title">{this.state.title}</h1>
                    <h1 className="survey-description">{this.state.description}</h1>
                    <SurveyLink link={document.location.href}/>
                    {!this.state.open && !this.admin_secret &&
                        <div className='results-info'>
                            <div className='aligned-center'>
                                This survey has been locked by the Survey Administrator. No new questions or votes can be submitted at this time.
                            </div>
                        </div>
                    }
                </div>
                <div>
                    {this.showResultsSummary() &&
                        <SurveyResultsInfo adminSecret={this.admin_secret} 
                                            surveyKey={this.surveyKey} 
                                            viewsNumber={this.state.viewsNumber}
                                            questionersNumber={this.state.questionersNumber}
                                            votersNumber={this.state.votersNumber} date={this.state.date} 
                                            open={this.state.open}
                                            surveyStateCallback={this.updateSurveyStateCallback}/>

                    }
                    {this.showNewQuestionBox() &&
                        <NewQuestionBox surveyKey={this.surveyKey} afterSubmit={this.addQuestionToState}/>
                    }
                </div>
                <div className="responses">
                    {
                        this.state.fetchInProgress ? <Loader type="TailSpin" color="#000000" height={80} width={80}/> :
                            this.getQuestionsLists()
                    }
                </div>
            </div>
        );
    }

    private showResultsSummary(): boolean {
        return Boolean(this.results_secret || this.admin_secret);
    }

    private showNewQuestionBox() {
        return !this.results_secret && !this.admin_secret && this.state.open;
    }
}
