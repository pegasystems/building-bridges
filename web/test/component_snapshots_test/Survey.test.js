// Link.react.test.js
import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Survey from '../../src/Pages/Survey/Survey';
import renderer from 'react-test-renderer';
import * as Models from '../../src/Models';

Enzyme.configure({ adapter: new Adapter() });

global.fetch = jest.fn(() =>
    Promise.resolve({
        json: () => Promise.resolve({})
    })
);

test('SurveyComponent renders properly test', () => {
    const component = renderer.create(
        <Survey/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});


global.flushPromises = () => new Promise(resolve => setImmediate(resolve));

global.fetch = jest.fn(() =>
    Promise.resolve({
        json: () => Promise.resolve({questions: [
                {_id: '1234', content: 'Question 1', upvotes: 1, downvotes: 0, isAuthor: true},
                {_id: '1235', content: 'Question 2', upvotes: 0, downvotes: 1, isAuthor: false}
            ], key: 'survey-1', date: '2020-06-22', title: 'Survey test'})
    })
);

test('SurveyComponent contains proper title test', () => {
    const wrapper = mount(
        <Survey match={{
            'params': {
                surveyKey: 'survey-1'
            }
        }} location={''}/>
    );
    return flushPromises().then(() => {
        wrapper.update();
        expect(wrapper.contains(<h1 className="survey-title">Survey test</h1>)).toEqual(true);
    });
});

test('Mark as read callback test', () => {
    const survey = new Survey();
    survey.state = {
        questions: [],
        fetchInProgress: true,
        title: '',
        hideVotes: false,
    };;
    survey.markAsReadCallback(Models.Question.createQuestionFromApiResult(
        {
            "content" : "content",
            "downvotes" : 0,
            "upvotes" : 0,
            "isAuthor" : 'true',
            "voted" : true,
            "_id" : "id",
            "read" : "false"
        }));

});