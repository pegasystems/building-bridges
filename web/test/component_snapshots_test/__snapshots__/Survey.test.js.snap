// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyComponent renders properly test 1`] = `
<div
  className="not-hide"
>
  <div
    className="title-div"
  >
    <h1
      className="survey-title"
    >
      
    </h1>
    <h1
      className="survey-description"
    >
      
    </h1>
    <i
      className="copy"
      onClick={[Function]}
    >
      Copy link
    </i>
  </div>
  <div>
    <div
      className="question-box"
    >
      <h4>
        Ask a question
      </h4>
      <form
        onSubmit={[Function]}
      >
        <textarea
          id="questionBox"
          name="question"
          onChange={[Function]}
          value=""
        />
        <label
          htmlFor="questionBox"
        >
          
        </label>
        <input
          className="add-new-button"
          type="submit"
          value="Submit"
        />
      </form>
    </div>
  </div>
  <div
    className="responses"
  >
    <div
      aria-busy="true"
      className=""
      style={Object {}}
    >
      <svg
        aria-label="audio-loading"
        height={80}
        viewBox="0 0 38 38"
        width={80}
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <linearGradient
            id="a"
            x1="8.042%"
            x2="65.682%"
            y1="0%"
            y2="23.865%"
          >
            <stop
              offset="0%"
              stopColor="#000000"
              stopOpacity="0"
            />
            <stop
              offset="63.146%"
              stopColor="#000000"
              stopOpacity=".631"
            />
            <stop
              offset="100%"
              stopColor="#000000"
            />
          </linearGradient>
        </defs>
        <g
          fill="none"
          fillRule="evenodd"
        >
          <g
            transform="translate(1 1)"
          >
            <path
              d="M36 18c0-9.94-8.06-18-18-18"
              id="Oval-2"
              stroke="#000000"
              strokeWidth="2"
            >
              <animateTransform
                attributeName="transform"
                dur="0.9s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </path>
            <circle
              cx="36"
              cy="18"
              fill="#fff"
              r={1}
            >
              <animateTransform
                attributeName="transform"
                dur="0.9s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </circle>
          </g>
        </g>
      </svg>
    </div>
  </div>
</div>
`;
